### PR TITLE
resolve: fix varlink message verification

### LINKED ISSUE #30199: resolved: Return parameters for method reply DumpCache() didn't pass validation on field 'dump', ignoring: No anode
Looks like `resolved` emits those messages every time `resolvectl show-cache` is run:
```sh
-bash-5.2# resolvectl flush-caches
-bash-5.2# resolvectl query H.local
...
-bash-5.2# resolvectl show-cache
Scope protocol=mdns family=AF_INET6 ifindex=2 ifname=host0
H.local IN AAAA fe80::741a:afff:fe88:1910

Scope protocol=mdns family=AF_INET ifindex=2 ifname=host0
H.local IN A 192.168.238.225
...
```
```sh
-bash-5.2# journalctl -u systemd-resolved --no-pager -e -o cat
...
varlink-26: New incoming message: {"method":"io.systemd.Resolve.Monitor.DumpCache","parameters":{}}
varlink-26: Changing state idle-server → processing-method
Mandatory field 'key' is null or missing on object, refusing.
Return parameters for method reply DumpCache() didn't pass validation on field 'dump', ignoring: No anode
varlink-26: Sending message: {"parameters":{"dump":[{"protocol":"mdns","family":10,"ifindex":2,"ifname":"host0","cache":[{"key":{"class":1,"type":28,"name":"H
.local"},"rrs":[{"rr":{"key":{"class":1,"type":28,"name":"H.local"},"address":[254,128,0,0,0,0,0,0,116,26,175,255,254,136,25,16]},"raw":"AUgFbG9jYWwAABwAAQAAA
HgAEP6AAAAAAAAAdBqv//6IGRA="}],"until":21976474407}]},{"protocol":"mdns","family":2,"ifindex":2,"ifname":"host0","cache":[{"key":{"class":1,"type":1,"name":"H
.local"},"rrs":[{"rr":{"key":{"class":1,"type":1,"name":"H.local"},"address":[192,168,238,225]},"raw":"AUgFbG9jYWwAAAEAAQAAAHgABMCo7uE="}],"until":21976472227
}]},{"protocol":"llmnr","family":2,"ifindex":2,"ifname":"host0","cache":[]},{"protocol":"dns","ifindex":2,"ifname":"host0","cache":[]},{"protocol":"llmnr","fa
mily":10,"ifindex":2,"ifname":"host0","cache":[]},{"protocol":"dns","cache":[]}]}}
varlink-26: Changing state processing-method → processed-method
varlink-26: Changing state processed-method → idle-server
```
Here's all the packets `resolved` sent/received:
```sh
    1   0.000000 192.168.238.232 → 224.0.0.251  MDNS 67 Standard query 0x0000 A H.local, "QM" question
    2   0.001226 192.168.238.225 → 224.0.0.251  MDNS 77 Standard query response 0x0000 A, cache flush 192.168.238.225
    3   0.008687 fe80::3430:b6ff:fe30:94d7 → ff02::fb     MDNS 87 Standard query 0x0000 AAAA H.local, "QM" question
    4   0.013514 fe80::741a:afff:fe88:1910 → ff02::fb     MDNS 109 Standard query response 0x0000 AAAA, cache flush fe80::741a:afff:fe88:1910
```

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #30199.

### CODE REVIEW

**Comment by evverx** [comment]
Thanks!

I guess I'd need to take a closer look. Even with this PR applied those messages are all over the place. For example here's what happens when the `io.systemd.Resolve.Monitor` interface is introspected:
```
varlink: New incoming connection.
varlink-25: Setting state idle-server
varlink-25: New incoming message: {"method":"org.varlink.service.GetInterfaceDescription","parameters":{"interface":"io.systemd.Resolve.Monitor"}}
varlink-25: Changing state idle-server → processing-method
Mandatory field 'interface' is null or missing on object, refusing.
Return parameters for method reply GetInterfaceDescription() didn't pass validation on field 'interface', ignoring: No anode
```
(it seems to me that ideally the CI should be able to catch those things automatically because copy-pasting all the fields seems to be error-prone)

**Comment by evverx** [comment]
The CI seems to have failed because `journaltcl ... --grep` didn't find anything and returned 1 and that's probably where `set -e` combined with `set -o pipefail` kicked in.

**Comment by evverx** [comment]
Also it seems resolved dumps null RRs
```
Nov 26 13:05:55 H systemd-resolved[19469]: varlink-25: New incoming message: {"method":"io.systemd.Resolve.Monitor.DumpCache","parameters":{}}                
Nov 26 13:05:55 H systemd-resolved[19469]: varlink-25: Changing state idle-server → processing-method                                                         
Nov 26 13:05:55 H systemd-resolved[19469]: Mandatory field 'rr' is null or missing on object, refusing.                                                       
Nov 26 13:05:55 H systemd-resolved[19469]: Return parameters for method reply DumpCache() didn't pass validation on field 'dump', ignoring: No anode  
```
```sh
                                "rrs" : [
                                        {
                                                "rr" : null,
                                                "raw" : "BFRlc3QL...
...
                                        }
                                ],
                                "until" : 16486563863
                        },
```
but it's a separate issue.

**Comment by yuwata** [comment]
@evverx Thank you for the comments and suggestions. Updated. PTAL.

File: test/units/testsuite-post.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+set -eux
+set -o pipefail
+
+TEST_NUM=${1?}
+echo "Running post script for TEST-${TEST_NUM}"
+
+(! journalctl -q -o short-monotonic --grep "didn't pass validation" >> /failed)
```
Comment (keszybz) [inline test/units/testsuite-post.sh]: Please drop the space after the redirect.

**Review by keszybz — CHANGES_REQUESTED** [review]

File: test/units/testsuite-post.sh (hunk 1 above)
Comment (mrc0mmand) [inline test/units/testsuite-post.sh]: What about replacing the inline shell script in the `end.service` with a standalone script and adding this check into it? So we don't have to hook up another script to the test services (which is easily forgotten).

Comment (mrc0mmand) [reply test/units/testsuite-post.sh]: (Or add it directly to `check_results_common` and `check_result_{nspawn,qemu}_unittests`)

Comment (yuwata) [reply test/units/testsuite-post.sh]: Let's try with end.service.

**Review by bluca — APPROVED** [review]

**Pull request merged**